ufm-test-services/metamask/tests/prometheusUtils.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config'
+import fetch from 'node-fetch'
+import { Counter, Pushgateway } from 'prom-client'
+
+const txSuccessMetricName = 'metamask_tx_success'
+const txFailureMetricName = 'metamask_tx_failure'
+
+const txSuccessCounter = new Counter({
+  name: txSuccessMetricName,
+  help: 'A counter signifying the number of successful transactions sent with Metamask since last failure',
+})
+const txFailureCounter = new Counter({
+  name: txFailureMetricName,
+  help: 'A counter signifying the number of failed transactions sent with Metamask since last successful transaction',
+})
+
+export const getMetricValueFromPrometheus = async (metricName: string) => {
+  if (process.env.PROMETHEUS_SERVER_URL === undefined)
+    throw new Error('Please provied PROMETHEUS_SERVER_URL in .env')
+
+  const response = await fetch(
+    `${process.env.PROMETHEUS_SERVER_URL}/api/v1/query?query=${metricName}`
+  )
+  const reponseJson = (await response.json()) as {
+    status: string
+    data: { resultType: string; result: any[] }
+  }
+
+  if (
+    reponseJson &&
+    reponseJson.data &&
+    reponseJson.data.result &&
+    reponseJson.data.result.length > 0
+  ) {
+    return parseFloat(reponseJson.data.result[0].value[1])
+  } else {
+    console.warn(`No data found for metric ${metricName} in Prometheus`)
+    return undefined
+  }
+}
+
+export const setMetamaskTxCounter = async (
+  isSuccess: boolean,
+  valueToSetTo: number
+) => {
+  if (process.env.PROMETHEUS_PUSHGATEWAY_URL === undefined)
+    throw new Error('Please provied PROMETHEUS_PUSHGATEWAY_URL in .env')
```

Comment:
This is fine but I prefer using zod for this
```typescript
const prometheusGatewayUrl = z.string().url().parse(process.env.PROMETHEUS_PUSHGATEWAY_URL)
```
Or if you have multiple env variables
```
const env = z.object({
  PROMETHEUS_PUSHGATEWAY_URL: z.string().url()
  ...
}).parse(process.env)
```